xain_fl/config/schema.py

```diff
@@ -0,0 +1,355 @@
+"""This module provides helpers for reading and validating the TOML
+configuration.
+
+"""
+from collections import namedtuple
+import ipaddress
+from typing import Any, Mapping, NamedTuple, Type, TypeVar
+import urllib
+
+import idna
+from schema import And, Optional, Or, Schema, SchemaError, Use
+import toml
+
+
+def error(key: str, description: str) -> str:
+    """Return an error message for the given configuration item and
+    description of the expected value type.
+
+    Args:
+
+        key (str): key of the configuration item
+        description (str): description of the expected type of value
+            for this configuration item
+    """
+    return f"Invalid `{key}`: value must be {description}"
+
+
+def strictly_positive_integer(
```

Comment:
That's funny. In french _strictly positive_ unambiguously means `> 0` while _positive_ means `>= 0`. In english, _strictly positive_ [seems to be used as well](https://en.wikipedia.org/wiki/Strictly_positive_measure) but it seems that _positive_ [can also mean `> 0`](https://math.stackexchange.com/questions/18464/is-positive-the-same-as-non-negative). Maybe let's ask @finiteprods what the preferred terminology is?

Reply:
I think you both sum up the situation pretty well! I wish we could say "natural number" and put this to rest but even there, there's sometimes (a similar) disagreement ;-) In my experience, "positive" _usually_ means `> 0` since we tend to say "non-negative" for `>= 0`. If you feel passionate about this, you could keep `strictly_positive_integer` but rename the other one to `nonnegative_integer`, otherwise just go with @janpetschexain 's suggestion.

Reply:
Thanks @finiteprods! I applied your suggestion @janpetschexain :)